### PR TITLE
LWDEV-7635 Implemented.

### PR DESCRIPTION
**Pull request by dmitrysmall**
Please, see: https://lykkex.atlassian.net/browse/LWDEV-7635.

### CODE REVIEW

File: src/Lykke.Job.CandleHistoryWriter.Repositories/HistoryMigration/HistoryProviders/TradesSQLHistory/TradesSqlHistoryRepository.cs [hunk 1]
```diff
@@ -91,8 +91,15 @@ string searchToken
...
-                                });
+                                    IsStraight =
+                                        reader.GetString(1) == SearchToken // If the trade is straight or reverse.
+                                };
+
+                                // We must ignore trades with negative prices and\or volumes (if any).
+                                if (trade.Price > 0 &&
+                                    trade.Volume > 0 &&
+                                    trade.OppositeVolume > 0)
+                                    result.Add(trade);
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.CandleHistoryWriter.Repositories/HistoryMigration/HistoryProviders/TradesSQLHistory/TradesSqlHistoryRepository.cs]: Add warning loging and increment health indicator on else

Comment (dmitrysmall) [reply src/Lykke.Job.CandleHistoryWriter.Repositories/HistoryMigration/HistoryProviders/TradesSQLHistory/TradesSqlHistoryRepository.cs]: Implemented logging. But for being able to increment a healtch counter, we have to inject migration health service into the sql repository and make some else unwanted manipulations. Znd from the other side, our log will now contain all the worings about negative trades, which might be enough for analysis. Are you sure this healt update is actually needed?

File: src/Lykke.Job.CandleHistoryWriter.Repositories/HistoryMigration/HistoryProviders/TradesSQLHistory/TradesSqlHistoryRepository.cs [hunk 2]
```diff
@@ -37,9 +38,9 @@ string searchToken
             )
         {
             _sqlQueryBatchSize = sqlQueryBatchSize;
-            _sqlConnString = sqlConnString;
+            _sqlConnString = !string.IsNullOrWhiteSpace(sqlConnString) ? sqlConnString : throw new ArgumentNullException(nameof(sqlConnString));
             _sqlTimeout = sqlTimeout;
-            _log = log;
+            _log = log.CreateComponentScope(nameof(TradesSqlHistoryRepository)) ?? throw new ArgumentNullException(nameof(log));
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.CandleHistoryWriter.Repositories/HistoryMigration/HistoryProviders/TradesSQLHistory/TradesSqlHistoryRepository.cs]: You've missed ? after log. And better to do such checks separately.

File: src/Lykke.Job.CandleHistoryWriter.Repositories/HistoryMigration/HistoryProviders/TradesSQLHistory/TradesSqlHistoryRepository.cs (hunk 1 above)
Comment (KonstantinRyazantsev) [reply src/Lykke.Job.CandleHistoryWriter.Repositories/HistoryMigration/HistoryProviders/TradesSQLHistory/TradesSqlHistoryRepository.cs]: If it's hard to implement, we can omit this. But health statistics is much more easier to monitor. Warning message could be missed in the flow of messages.

**Pull request merged**